zeth-contracts/contracts/BaseMixer.sol

```diff
@@ -98,20 +126,122 @@ contract BaseMixer is MerkleTreeMiMC7, ERC223ReceivingContract {
         emit LogMerkleRoot(initialRoot);
 
         token = token_address;
+
+        // We require the need of a single residual field elements
+        require(
+            field_capacity < digest_length,
+            "A hash digest fits in a single field element."
+        );
+        require(
+            length_bit_residual < field_capacity,
+            "Too many input and output notes considered."
+        );
     }
 
     // ============================================================================================ //
     // Reminder: Remember that the primary inputs are ordered as follows:
     // We make sure to have the primary inputs ordered as follow:
-    // [Root, NullifierS, CommitmentS, value_pub_in, value_pub_out]
+    // [Root, NullifierS, CommitmentS, h_sig, h_iS, Residual Field Element(S)]
     // ie, below is the index mapping of the primary input elements on the protoboard:
     // - Index of the "Root" field elements: {0}
     // - Index of the "NullifierS" field elements: [1, NumInputs + 1[
     // - Index of the "CommitmentS" field elements: [NumInputs + 1, NumOutputs + NumInputs + 1[
```

Comment:
`// - Index of the "CommitmentS" field elements: [1+ NumInputs, 1 + NumInputs + NumOutputs[`

Reply:
Fixed